demos/shapes-custom-question-authoring-demo/src/question/index.js

```diff
@@ -0,0 +1,339 @@
+                    }
+                })
+                shapeButton.classList.toggle('selected');
+                shapeButton.parentElement.style.outlineColor = question.selected_shape_color;
+                
+                response = shapeButton.getAttribute('data-index');
+                events.trigger('changed', response);
+                facade.resetValidationUI();
+            })
+        })
+
+
+        // "validate" event can be triggered when Check Answer button is clicked or when public method .validate() is called
+        // so developer needs to listen to this event to decide if he wants to display the correct answers to user or not
+        // options.showCorrectAnswers will tell if correct answers for this question should be display or not.
+        // The value showCorrectAnswers by default is the value of showCorrectAnswers inside initOptions object that is used
+        // to initialize question app or the value of the options that is passed into public method validate (like question.validate({showCorrectAnswers: false}))
+        events.on('validate', options => {
+            // OPTIONAL Step 1: 
+            //If you want to show changes to the UI for a correct or incorrect answer when the learner presses check answer
+            // then make sure you have also implemented facade.showValidationUI(), and that you call it here:
+            facade.showValidationUI()
+
+            // OPTIONAL Step 2: 
+            // If you want to display the correct answer to the learner when they press the Check Answer button,
+            // then you should leverage the suggestedAnswersList.setAnswers() method. Please see the example implementation below for full details.
+
+            // EXAMPLE Implementation 
+            
+            if (!facade.isValid() && options.showCorrectAnswers) {
+                // pass a string if there is a single correct answer 
+                //OR:
+                /**
+                * For custom questions with multiple correct answers, you can also pass an array of objects into this method, 
+                * each object containing an index key of type number for the 0-based index of the correct answer, 
+                * and a label key of type string for the text of the correct answer. 
+                * So, for a hypothetical question with 3 correct answers, you might pass the following array into setAnswers():
+
+                    [
+                     {index: 0, label: "correct answer 1"},
+                     {index: 1, label: "correct answer 2"},
+                     {index: 2, label: "correct answer 3"}
+                    ]
+
+                 */
+                
+                // EXAMPLE Implementation for a correct answer that is a string:
+                
+                this.suggestedAnswersList.setAnswers([{index: Number(question.validation.valid_response.value), label: question.shapes[Number(question.validation.valid_response.value)].kind.toUpperCase()}]);
```

Comment:
Add a null safe to the `question.validation.valid_response` and `question.shapes` before we add them to answerList
```suggestion
                if (question.validation?.valid_response?.value != null) {
                    const validResponseIndex = Number(question.validation.valid_response.value);
                    
                    if (question.shapes?.[validResponseIndex]) {
                        const correctShape = question.shapes[validResponseIndex];

                        this.suggestedAnswersList.setAnswers([{
                            index: validResponseIndex,
                            label: correctShape.kind.toUpperCase()
                        }]);
                    }
                }
```